### PR TITLE
subsys: lorawan: add devicetime request support

### PR DESCRIPTION
**Pull request by kowalewskijan**
This is a hijacked PR from https://github.com/zephyrproject-rtos/zephyr/pull/55073

According to comments vs. what was available on: https://github.com/RomainPelletant/zephyr/tree/subsys/lorawan/add-devicetime-request missing parts were:

- [x] Using GPS epoch instead of UNIX
- [x] Removing redundant `lorawan_req_time` helper function
- [x] Changing `data_pending` bool flag to `uint8_t flags` field in `multicast_package_callback` function

@martinjaeger @JordanYates @RomainPelletant - please let me know if there is anything I missed from the original PR discussion.

### CODE REVIEW

**Comment by RomainPelletant** [comment]
A clang format run seems to be required. Except that: LGTM.
Thank you @kowalewskijan

**Comment by kowalewskijan** [comment]
All issues resolved - CI is green

**Comment by kowalewskijan** [comment]
Bump @JordanYates @martinjaeger

File: subsys/lorawan/lorawan.c [hunk 1]
```diff
@@ -383,6 +392,42 @@ int lorawan_set_region(enum lorawan_region region)
...
+	return ret;
+}
+
+int lorawan_device_time_get(uint32_t *gps_time)
+{
+	SysTime_t local_time;
+
+	__ASSERT(gps_time != NULL, "gps_time parameter is required");
+
+	if (lorawan_device_time_updated_once) {
```
Comment (JordanYates) [inline subsys/lorawan/lorawan.c]: Minor style suggestion:
```
if (!updated_once) {
    return -EAGAIN;
}
/* Do work here */
return 0;
```

Comment (kowalewskijan) [reply subsys/lorawan/lorawan.c]: fixed

**Review by JordanYates — APPROVED** [review]

**Review by mniestroj — APPROVED** [review]

**Pull request merged**